packages/expo-constants/ios/ConstantsModule.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  ConstantsModule.swift
+//  ExpoConstants-ExpoConstants
+//
+//  Created by Alan Hughes on 11/11/2022.
+//
+
+import ExpoModulesCore
+import WebKit
+
+public class ConstantsModule: Module {
+
+    private lazy var constants = {
+        let dict = appContext?.constants?.constants()
+
+        if let result = dict as? [String: Any?] {
+            return result
+        }
+
+        return [String: Any?]()
+    }()
+
+    public func definition() -> ModuleDefinition {
+        Name("ExponentConstants")
+
+        Constants {
+            return constants
+        }
+
+        AsyncFunction("getWebViewUserAgentAsync") { (promise: Promise) in
+            let webView = WKWebView()
+            let result = webView.value(forKey: "userAgent")
+            promise.resolve(result)
+        }.runOnQueue(.main)
```

Comment:
```suggestion
        AsyncFunction("getWebViewUserAgentAsync") {
            let webView = WKWebView()
            return webView.value(forKey: "userAgent")
        }.runOnQueue(.main)
```

Reply:
This results in a compilation error `Generic parameter 'R' could not be inferred`. value(forKey:) is returning Any?. Casting doesn't seem to help. 

AsyncFunction("getWebViewUserAgentAsync") { () -> Any? in
How about this?

Reply:
Hmm, I think `return webView.value(forKey: "userAgent") as? String` should work 🤔

Reply:
No, that won't work either. I don't like the Any? either but not sure what's best to

Reply:
Interesting, looks like the compiler is not smart enough and we need to tell him explicitly what's the return type 😅 
Try this then:
```swift
AsyncFunction("getWebViewUserAgentAsync") { () -> String? in
  let webView = WKWebView()
  return webView.value(forKey: "userAgent") as? String
}.runOnQueue(.main)
```

Reply:
Yes, that'll work 👍